File: 168.excel-sheet-column-title.py
#
# @lc app=leetcode id=168 lang=python3
#
# [168] Excel Sheet Column Title
#

# Time:  O(logn)
# Space: O(1)

# @lc code=start
class Solution:
    def convertToTitle(self, columnNumber: int) -> str:
        res = ""
        while columnNumber:
            res = chr((columnNumber - 1) % 26 + 65) + res
            columnNumber = (columnNumber - 1) // 26
        return res


# @lc code=end

File: 168.excel-sheet-column-title.go
/*
 * @lc app=leetcode id=168 lang=golang
 *
 * [168] Excel Sheet Column Title
 */

package leetcode

// @lc code=start
func convertToTitle(columnNumber int) string {
	res := []byte{}
	for columnNumber > 0 {
		res = append(res, 'A'+byte((columnNumber-1)%26))
		columnNumber = (columnNumber - 1) / 26
	}
	for i := 0; i < len(res)/2; i++ {
		res[i], res[len(res)-1-i] = res[len(res)-1-i], res[i]
	}
	return string(res)
}

// @lc code=end
